ciphers/hill_cipher.py

```diff
@@ -129,32 +166,55 @@ def make_decrypt_key(self):
         return self.toInt(self.modulus(inv_key))
 
     def decrypt(self, text):
+        """
+        >>> HillCipher(numpy.matrix([[2,5],[1,6]])).decrypt('WHXYJOLM9C6XT085LL')
+        'TESTINGHILLCIPHERR'
+        """
         self.decrypt_key = self.make_decrypt_key()
         text = self.process_text(text.upper())
         decrypted = ""
 
         for i in range(0, len(text) - self.break_key + 1, self.break_key):
             batch = text[i : i + self.break_key]
-            batch_vec = list(map(self.replaceLetters, batch))
+            batch_vec = list(map(HillCipher(self.encrypt_key).replace_letters, batch))
             batch_vec = numpy.matrix([batch_vec]).T
             batch_decrypted = self.modulus(self.decrypt_key.dot(batch_vec)).T.tolist()[
                 0
             ]
-            decrypted_batch = "".join(list(map(self.replaceNumbers, batch_decrypted)))
+            decrypted_batch = "".join(
+                list(map(HillCipher(self.encrypt_key).replace_numbers, batch_decrypted))
+            )
             decrypted += decrypted_batch
 
         return decrypted
 
 
-def main():
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+    """
+    Enter the order of the encryption key: 2
+Enter each row of the encryption key with space separated integers
+'2 5'
+'1 6'
+Would you like to encrypt or decrypt some text? (1 or 2)
+
+1. Encrypt
+2. Decrypt
+1
+What text would you like to encrypt?: 'testing hill cipher'
+Your encrypted text is:
+WHXYJOLM9C6XT085LL
+"""
     N = int(input("Enter the order of the encryption key: "))
-    hill_matrix = []
+    hill_matrix: list = []
```

Comment:
The type hint is not really required here because CPython is smart enough to know that [] is a list.  A useful type hint would be `hill_matrix: List[int] = []` assuming that `from typing import List` was added above.